PWGJE/Core/JetTaggingUtilities.h

```diff
@@ -305,19 +323,48 @@ int16_t getJetFlavor(AnyJet const& jet, AllMCParticles const& mcparticles)
       if (dR < jet.r() / 100.f) {
         if (TMath::Abs(pdgcode) == 5) {
           return 2; // Beauty jet
-        } else {
```

Comment:
Hi Hadi, whats the new logic on this part of the code?

Reply:
why isnt 1 returned for charm quark directly in the loop if the pdgcode is 4 instead of setting charmQuark=true

Reply:
This copied from AliPhysics with modification. We can't return a charm if finds it in the stack. Because it could be that there is a case in which the charm appears first in the stack and then comes the beauty, then this will be mistaken as a charm. To be checked later if this is really the case in O2.